File: student-app/StarIsle-student/lib/screens/profile_screen.dart
/// @file profile_screen.dart
/// @description 学生端"我的"页面，展示个人信息卡片、隐私与安全入口、帮助中心与设置项，
///              通过命名路由跳转到对应子页面。
/// @module student-app/screens

import 'package:flutter/material.dart';

/// 学生端个人中心页面。
class ProfileScreen extends StatelessWidget {
  /// 构造函数。
  const ProfileScreen({super.key});

  /// 构建页面主体。
  ///
  /// 返回：包含个人信息、隐私与安全、帮助中心、设置等卡片的 [Scaffold]。
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      // 顶部 AppBar：标题"我的"
      appBar: AppBar(
        title: const Text('我的'),
      ),
      body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          // 个人信息卡片：头像 + 昵称 + 身份 + 编辑入口
          Card(
            child: Padding(
              padding: const EdgeInsets.all(16),
              child: Row(
                children: [
                  // 用户头像
                  CircleAvatar(
                    radius: 32,
                    backgroundColor: Theme.of(context).primaryColor,
                    child: const Icon(
                      Icons.person,
                      size: 32,
                      color: Colors.white,
                    ),
                  ),
                  const SizedBox(width: 16),
                  // 昵称与身份信息
                  const Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          '小明',
                          style: TextStyle(
                            fontSize: 18,
                            fontWeight: FontWeight.w600,
                          ),
                        ),
                        SizedBox(height: 4),
                        Text(
                          '高中生',
                          style: TextStyle(
                            fontSize: 14,
                            color: Colors.grey,
                          ),
                        ),
                      ],
                    ),
                  ),
                  // 编辑个人信息按钮
                  IconButton(
                    icon: const Icon(Icons.edit),
                    onPressed: () {
                      // TODO: 编辑个人信息
                    },
                  ),
                ],
              ),
            ),
          ),

          const SizedBox(height: 16),

          // 隐私与安全入口
          Card(
            child: ListTile(
              leading: const Icon(Icons.security),
              title: const Text('隐私与安全'),
              subtitle: const Text('查看数据收集清单、导出/删除数据'),
              trailing: const Icon(Icons.arrow_forward_ios),
              onTap: () {
                Navigator.pushNamed(context, '/privacy_settings');
              },
            ),
          ),

          const SizedBox(height: 16),

          // 帮助中心卡片：危机资源与使用指南
          Card(
            child: Column(
              children: [
                // 危机资源入口
                ListTile(
                  leading: const Icon(Icons.help_outline),
                  title: const Text('危机资源'),
                  subtitle: const Text('心理援助热线'),
                  trailing: const Icon(Icons.arrow_forward_ios),
                  onTap: () {
                    Navigator.pushNamed(context, '/crisis_resources');
                  },
                ),

                // 使用指南入口
                ListTile(
                  leading: const Icon(Icons.book_outlined),
                  title: const Text('使用指南'),
                  subtitle: const Text('如何使用星屿'),
                  trailing: const Icon(Icons.arrow_forward_ios),
                  onTap: () {
                    Navigator.pushNamed(context, '/user_guide');
                  },
                ),
              ],
            ),
          ),

          const SizedBox(height: 16),

          // 设置卡片：通知、深色模式、关于
          Card(
            child: Column(
              children: [
                // 通知设置入口
                ListTile(
                  leading: const Icon(Icons.notifications_outlined),
                  title: const Text('通知设置'),
                  trailing: const Icon(Icons.arrow_forward_ios),
                  onTap: () {
                    // TODO: 打开通知设置
                  },
                ),

                // 深色模式入口
                ListTile(
                  leading: const Icon(Icons.dark_mode),
                  title: const Text('深色模式'),
                  trailing: const Icon(Icons.arrow_forward_ios),
                  onTap: () {
                    // TODO: 打开深色模式设置
                  },
                ),

                // 关于星屿入口
                ListTile(
                  leading: const Icon(Icons.info_outline),
                  title: const Text('关于星屿'),
                  trailing: const Icon(Icons.arrow_forward_ios),
                  onTap: () {
                    Navigator.pushNamed(context, '/about');
                  },
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

File: student-app/StarIsle-student-android/app/src/main/java/com/starisle/student/ui/screens/ProfileScreen.kt
package com.starisle.student.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.Book
import androidx.compose.material.icons.filled.DarkMode
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.HelpOutline
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Security
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.navigation.NavController
import com.starisle.student.theme.StarNightBlue

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ProfileScreen(navController: NavController? = null) {
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("我的") },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = StarNightBlue,
                    titleContentColor = Color.White,
                ),
            )
        },
    ) { padding ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp),
        ) {
            // 个人信息卡片
            item {
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(16.dp),
                ) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(16.dp),
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        // 头像
                        IconButton(
                            onClick = {},
                            modifier = Modifier.size(64.dp),
                        ) {
                            Icon(
                                Icons.Filled.Person,
                                contentDescription = null,
                                tint = Color.White,
                                modifier = Modifier.size(32.dp),
                            )
                        }
                        Spacer(Modifier.width(16.dp))
                        Column(modifier = Modifier.weight(1f)) {
                            Text("小明", fontSize = 18.sp, fontWeight = FontWeight.W600)
                            Spacer(Modifier.height(4.dp))
                            Text(
                                "高中生",
                                fontSize = 14.sp,
                                color = Color.Gray,
                            )
                        }
                        IconButton(onClick = { /* TODO: 编辑个人信息 */ }) {
                            Icon(Icons.Filled.Edit, contentDescription = "编辑")
                        }
                    }
                }
            }

            // 隐私与安全入口
            item {
                ProfileEntryCard(
                    icon = Icons.Filled.Security,
                    title = "隐私与安全",
                    subtitle = "查看数据收集清单、导出/删除数据",
                    onClick = { /* TODO: /privacy_settings */ },
                )
            }

            // 帮助中心
            item {
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(16.dp),
                ) {
                    Column {
                        ProfileRow(
                            icon = Icons.Filled.HelpOutline,
                            title = "危机资源",
                            subtitle = "心理援助热线",
                            onClick = { /* TODO: /crisis_resources */ },
                        )
                        ProfileRow(
                            icon = Icons.Filled.Book,
                            title = "使用指南",
                            subtitle = "如何使用星屿",
                            onClick = { /* TODO: /user_guide */ },
                        )
                    }
                }
            }

            // 设置
            item {
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(16.dp),
                ) {
                    Column {
                        ProfileRow(
                            icon = Icons.Filled.Notifications,
                            title = "通知设置",
                            onClick = { /* TODO: 通知设置 */ },
                        )
                        ProfileRow(
                            icon = Icons.Filled.DarkMode,
                            title = "深色模式",
                            onClick = { /* TODO: 深色模式 */ },
                        )
                        ProfileRow(
                            icon = Icons.Filled.Info,
                            title = "关于星屿",
                            onClick = { /* TODO: /about */ },
                        )
                    }
                }
            }
        }
    }
}

@Composable
private fun ProfileEntryCard(
    icon: ImageVector,
    title: String,
    subtitle: String,
    onClick: () -> Unit,
) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        onClick = onClick,
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Icon(icon, contentDescription = null)
            Spacer(Modifier.width(16.dp))
            Column(modifier = Modifier.weight(1f)) {
                Text(title)
                Text(subtitle, fontSize = 12.sp, color = Color.Gray)
            }
            Icon(Icons.AutoMirrored.Filled.KeyboardArrowRight, contentDescription = null)
        }
    }
}

@Composable
private fun ProfileRow(
    icon: ImageVector,
    title: String,
    subtitle: String? = null,
    onClick: () -> Unit,
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(16.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(icon, contentDescription = null)
        Spacer(Modifier.width(16.dp))
        Column(modifier = Modifier.weight(1f)) {
            Text(title)
            if (subtitle != null) {
                Text(subtitle, fontSize = 12.sp, color = Color.Gray)
            }
        }
        Icon(Icons.AutoMirrored.Filled.KeyboardArrowRight, contentDescription = null)
    }
}
